.github/workflows/Pitests.yml

```diff
@@ -0,0 +1,639 @@
+        with:
+          path: ~/.m2/repository
+          key: checkstyle-maven-cache-${{ hashFiles('**/pom.xml') }}
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Pitest-whitespace
+        run: |
+          mkdir pitest-whitespace-report
+          ./.ci/pitest.sh pitest-whitespace
+      - name: Copy result
+        if: always()
+        run: cp -r target/pit-reports/ pitest-whitespace-report/
+      - name: Archive code coverage results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: pitest4-whitespace-coverage-report
+          path: /home/runner/work/checkstyle/checkstyle/pitest-whitespace-report/
+          retention-days: 3
+
+  Pitest-filters:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup local maven cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: checkstyle-maven-cache-${{ hashFiles('**/pom.xml') }}
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Pitest-filters
+        run: |
+          mkdir pitest-filters-report
+          ./.ci/pitest.sh pitest-filters
+      - name: Copy result
+        if: always()
+        run: cp -r target/pit-reports/ pitest-filters-report/
+      - name: Archive code coverage results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: pitest4-filters-coverage-report
+          path: /home/runner/work/checkstyle/checkstyle/pitest-filters-report/
+          retention-days: 3
+
+  Pitest-header:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup local maven cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: checkstyle-maven-cache-${{ hashFiles('**/pom.xml') }}
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Pitest-header
+        run: |
+          mkdir pitest-header-report
+          ./.ci/pitest.sh pitest-header
+      - name: Copy result
+        if: always()
+        run: cp -r target/pit-reports/ pitest-header-report/
+      - name: Archive code coverage results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: pitest4-header-coverage-report
+          path: /home/runner/work/checkstyle/checkstyle/pitest-header-report/
+          retention-days: 3
+
+  Pitest-common-2:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup local maven cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: checkstyle-maven-cache-${{ hashFiles('**/pom.xml') }}
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: pitest-common-2
+        run: |
+          mkdir pitest-common-2-report
+          ./.ci/pitest.sh pitest-common-2
+      - name: Copy result
+        if: always()
+        run: cp -r target/pit-reports/ pitest-common-2-report/
+      - name: Archive code coverage results
+        if: always()
```

Comment:
there are still these `always()`, please remove all of them

Reply:
These are `always()` are in steps, not like previously which were on jobs. It is here to make sure that even if test fail, we can upload reports if any. These are not `true` by default, and must be manually set.